extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala

```diff
@@ -131,4 +140,39 @@ object SparkRangerAdminPlugin extends RangerBasePlugin("spark", "sparkSql")
       }
     }
   }
+
+  def getOrCreateRangerPlugin(catalog: Option[String] = None): RangerBasePlugin = synchronized {
+    catalog match {
+      case None | Some("spark_catalog") =>
+        defaultBasePlugin
+      case Some(catalogName) =>
+        val serviceName = getRangerConf(defaultBasePlugin)
+          .get(s"ranger.plugin.spark.catalog.$catalogName.service.name")
+        serviceName match {
+          case _ if StringUtils.isBlank(serviceName) =>
+            logWarning(s"config ranger.plugin.spark.catalog.$catalogName.service.name not found," +
+              s" default ranger plugin is used")
+            defaultBasePlugin
+          case _ =>
+            catalog2pluginMap.getOrElseUpdate(
+              catalogName,
+              initRangerBasePlugin(serviceName = serviceName, appId = catalogName))
+        }
+    }
+  }
+
+  def initRangerBasePlugin(serviceName: String, appId: String): RangerBasePlugin = {
+    val basePlugin =
+      if (isRanger21orGreater) {
+        classOf[RangerBasePlugin].getConstructor(classOf[String], classOf[String], classOf[String])
+          .newInstance(serviceType, serviceName, appId)
+      } else {
+        // ignoring serviceName for Ranger 2.0 and below,
```

Comment:
Explain why not what

Reply:
Since it's not supporting different service names with same service type in Ranger 2.0 and below.
The only way to separate plugin config with service type is to seperate them in `ranger-security-${serviceType}.xml` configs file with different service types, plus repeating all the detail configs in it (e.g. REST url, user, password, and etc.). This is too heavy and hard to use.